### PR TITLE
FAI-342 - Make LIME parameters configurable in one place

### PR DESCRIPTION
**Pull request by tteofili**
Moving all LIME  parameters into one place allows for better user experience, awareness of the different factors impacting explanations and, eventually, feed them into an optimization pipeline (e.g. seeking impact-score, stability, etc.).

See https://issues.redhat.com/browse/FAI-342

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by tteofili** [comment]
jenkins retest this please

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeConfig.java [hunk 1]
```diff
@@ -95,4 +118,27 @@ public boolean adaptDatasetVariance() {
...
+
+    public double getProximityKernelWidth() {
+        return proximityKernelWidth;
+    }
+
+    public double getEncodingClusterThreshold() {
+        return encodingClusterThreshold;
+    }
+
+    public LimeConfig withEncodingClusterThreshold(double encodingClusterThreshold) {
```
Comment (jiripetrlik) [inline explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeConfig.java]: Would it be possible to create a test which will use all these setters in chain and then check using getters whether the value was set properly?

**Review by jiripetrlik — COMMENTED** [review]
I have one suggestion related to tests.

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/EncodingParams.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Params for {@link Type}-aware encoding.
+ *
+ * See {@link Type#encode(EncodingParams, Value, Value[])}
+ */
+public class EncodingParams {
+
+    private final double numericTypeClusterGaussianFilterWidth;
+    private final double umericTypeClusterThreshold;
```
Comment (r00ta) [inline explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/EncodingParams.java]: typo :)

**Review by r00ta — APPROVED** [review]
:+1:

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/EncodingParams.java (hunk 2 above)
Comment (tteofili) [reply explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/EncodingParams.java]: lol, thx

**Review by jiripetrlik — APPROVED** [review]
@tteofili 
Thank you for additional test. LGTM now.

**Review by r00ta — APPROVED** [review]

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/DatasetEncoder.java [hunk 3]
```diff
@@ -41,13 +42,15 @@
     private final List<Output> predictedOutputs;
     private final List<Feature> targetInputFeatures;
     private final Output originalOutput;
+    private final EncodingParams encodingParams;
 
     DatasetEncoder(List<PredictionInput> perturbedInputs, List<Output> perturbedOutputs,
-                   List<Feature> targetInputFeatures, Output targetOutput) {
+                   List<Feature> targetInputFeatures, Output targetOutput, EncodingParams encodingParams) {
```
Comment (danielezonca) [inline explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/DatasetEncoder.java]:
```suggestion
                   List<Feature> targetInputFeatures, Output targetOutput, 
                   EncodingParams encodingParams) {
```

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeExplainer.java [hunk 4]
```diff
@@ -179,13 +180,15 @@ private void getSaliency(List<Feature> linearizedTargetInputFeatures, Map<String
         List<FeatureImportance> featureImportanceList = new LinkedList<>();
 
         // encode the training data so that it can be fed into the linear model
+        EncodingParams encodingParams = new EncodingParams(limeConfig.getEncodingGaussianFilterWidth(), limeConfig.getEncodingClusterThreshold());
         DatasetEncoder datasetEncoder = new DatasetEncoder(limeInputs.getPerturbedInputs(),
-                                                           limeInputs.getPerturbedOutputs(),
-                                                           linearizedTargetInputFeatures, originalOutput);
+                limeInputs.getPerturbedOutputs(),
+                linearizedTargetInputFeatures, originalOutput, encodingParams);
```
Comment (danielezonca) [inline explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeExplainer.java]: What about add a `getEncodingParams()` method to `LimeConfig` so here the code become
```suggestion
                linearizedTargetInputFeatures, originalOutput, 
                limeConfig.getEncodingParams());
```

**Review by danielezonca — APPROVED** [review]

**Comment by danielezonca** [comment]
jenkins retest this please

**Pull request merged**